spec/support/json_controller_helper.rb

```diff
@@ -1,91 +1,93 @@
+    attributes_for "updated_valid_#{factory_name}".to_sym
   end
 
   # set of invalid attributes required to update model
   let(:updated_invalid_attributes) do
-    attributes_for "invalid_#{model_string}".to_sym
+    attributes_for "invalid_#{factory_name}".to_sym
   end
 
   # symbol of the variable being assigned by the
   # controller in all the other methods
-  let(:assigned_variable) { model_symbol }
+  let(:assigned_variable) { factory }
 
   # symbol of the variable being assigned by the RESTful
   # controller in the index method
-  let(:mass_assigned_variable) { model_string.pluralize.to_sym }
+  let(:mass_assigned_variable) { factory_name.pluralize.to_sym }
+
+  let(:model) { FactoryGirl.create(factory) }
 
   describe 'GET index' do
     it 'assigns all models as @models' do
-      model = model_class.create! valid_attributes
-      get :index, request_params, valid_session
+      get :index, request_params(model), valid_session
       assigns(mass_assigned_variable).should eq([model])
     end
   end
 
   describe 'GET show' do
     it 'assigns the requested model as @model' do
-      model = model_class.create! valid_attributes
-      get :show, request_params(id: model.to_param), valid_session
+      get :show, request_params(model, id: model.to_param), valid_session
       assigns(assigned_variable).should eq(model)
     end
   end
 
   describe 'POST create' do
     describe 'with valid params' do
-      it 'creates a new model_class' do
+      it 'creates a new model' do
+        sample_model = FactoryGirl.build(factory)
         expect do
           post :create,
-               request_params(assigned_variable => valid_attributes),
+               request_params(nil, assigned_variable => valid_attributes),
                valid_session
-        end.to change(model_class, :count).by(1)
+        end.to change(sample_model.class, :count).by(1)
       end
 
-      it 'assigns a newly created course as @course' do
+      it 'assigns a newly created model as @model' do
+        sample_model = FactoryGirl.build(factory)
         post :create,
-             request_params(assigned_variable => valid_attributes),
+             request_params(nil, assigned_variable => valid_attributes),
              valid_session
-        assigns(assigned_variable).should be_a(model_class)
+        assigns(assigned_variable).should be_a(sample_model.class)
         assigns(assigned_variable).should be_persisted
       end
 
       it 'returns :created code' do
         post :create,
-             request_params(assigned_variable => valid_attributes),
+             request_params(nil, assigned_variable => valid_attributes),
              valid_session
         expect(response.response_code).to eq(code(:created))
       end
     end
 
     describe 'with invalid params' do
-      it 'assigns a newly created but unsaved course as @course' do
+      it 'assigns a newly created but unsaved model as @model' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.